### ISSUE TITLE
Road Map - API 15 Support

### ISSUE THREAD

**Issue by jdalrymple** [jdalrymple/gitbeaker#1702]
## What is this?
I figured it would be nice to have an overview of what is coming up next both for myself and anyone interested in the project. This issue outlines the issues I am focusing on, in order of priority and why. I will also mention any PR's that solve this issues along side them.

### Implement Browser Integration Tests ☑️

**Related Issues:** #1603

*tldr:* The browser release occasionally runs into compilation hiccups, and so after the last fix, it became quite apparent that tests should exist for this. Unfortunately, building proper integration tests that test against a Gitlab instance was a bit more complicated and as of right now, has me looking into doing some of the testing infrastructure in GCP.

### Improve Testing Infrastructure 

**Related Issues:**  ☑️ #1251 , #1625

**Related PR(s):** #1604

To improve release speed, proper testing should optimize these test stages.

This would look like:

```
install -> build -> startup Cloudrun Gitlab instance -> integration tests (making requests to cloud run)
        -> unit tests
        -> lint
```

To reduce the amount of GCP calls however, only specific PRs should trigger the integration tests. In this case, anyting that is not a minor, major or breaking change, should skip those tests. As an example: https://github.com/circleci/circleci-docs/blob/master/jekyll/_cci2/workflows.md

### Overhaul Typing

**Related Issues:** #1088 #1306 #1482 #1218 #519 #492

**Related PR:**  #1768 

*tldr:* Adding all the types for every services response. This will be helpful for documentation during development AND [auto gen docs](#39)

### Update API Support

**Related Issues:** #1623  #1620 #1763

*tldr:* Update the library to match the latest Gitlab API (14). I'm not sure if i will tackle this with # 2, because I'm seeing all the missing info as I add the types, but it should be done and released all at the same time to avoid confusion

## Timeline
Currently on hiatus until I finish some personal life things. Any help would be appreciated though! On going, testing is taking a while

**Comment by jdalrymple** [comment]
Latest canary release is out! -> 36.0.2--canary.2258.813948675.0
